pkg/ko/build/gobuild.go

```diff
@@ -249,8 +249,10 @@ func (gb *gobuild) Build(s string) (v1.Image, error) {
 	}
 	layers = append(layers, dataLayer)
 
+	appPath := filepath.Join(appDir, filepath.Base(s))
```

Comment:
Are there any cases here where `filepath.Base(s)` would be invalid (ie. return "." or "/")?

As far as I can tell `ko` should only work on well-defined importpaths inside GOPATH so this shouldn't be an issue.

Reply:
We'd like to make it work outside of GOPATH, I think, for go modules. It might make sense to check for "." or "/" explicitly and return a better error just in case we hit that path for some reason. It would make future debugging a lot easier.

Reply:
Alternatively, in that case we could just use the legacy app name and containers will continue to build.

Reply:
In the case that this happens now the entrypoint will default to "/ko-app/ko-app".